lib/Parse/ParseDecl.cpp

```diff
@@ -5728,6 +5728,15 @@ void Parser::skipSILUntilSwiftDecl() {
   }
 }
 
+void Parser::skipAnyAttribute() {
+  consumeToken(tok::at_sign);
+  if (!consumeIf(tok::identifier))
+    return;
+
+  if (consumeIf(tok::l_paren))
+    skipUntil(tok::r_paren);
```

Comment:
`skipUntil(tok::r_paren)` doesn't consume `)`. I'd suggest

```suggestion
  if (Tok.is(tok::l_paren))
    skipSingle();
```

Reply:
On second thought, this is ambiguous `foo { @someAttr (a: Int) -> Int in }`. We should do some heuristics just like: https://github.com/apple/swift/blob/e024d96f59ba26a336b2e88490385ba2a19375d7/lib/Parse/ParseDecl.cpp#L3178-L3186

Reply:
You're right, and if I'd written the tests @slavapestov rightly asked for I would have found this. Thank you! I'll clean this up shortly.